irohad/ametsuchi/impl/storage_impl.cpp

```diff
@@ -414,25 +419,23 @@ namespace iroha {
     }
 
     boost::optional<std::unique_ptr<LedgerState>> StorageImpl::commit(
-        std::unique_ptr<MutableStorage> mutableStorage) {
-      auto storage_ptr = std::move(mutableStorage);  // get ownership of storage
-      auto storage = static_cast<MutableStorageImpl *>(storage_ptr.get());
-      for (const auto &block : storage->block_store_) {
-        storeBlock(*block.second);
-      }
+        std::unique_ptr<MutableStorage> mutable_storage) {
+      auto storage = static_cast<MutableStorageImpl *>(mutable_storage.get());
```

Comment:
This line looks a bit unclear, could you explain please why it's necessary?

Reply:
It is used to access the private field of `MutableStorageImpl`:
https://github.com/hyperledger/iroha/blob/a5acbdc1737c680c6d15a18c31c305ac3a7d16f6/irohad/ametsuchi/impl/storage_impl.cpp#L425
Legacy design 🤷‍♂️